runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -2366,7 +2379,7 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          char *nextSignature = (char*)alloca(methodDescriptorLength+1);
```

Comment:
`nextSignatureString.data()` returns `const char*` which will not work with `getStringUTF8()` since it changes the content and a compilation error "`invalid conversion from ‘const char*’ to ‘char*`"  will throw too. Looking at other places where `getStringUTF8()` is called, `alloca` is called often. Either it could be left as it is or another alternative is to use `trMemory->allocateStackMemory` as https://github.com/eclipse/openj9/blob/fd96afe5f58b9125d812dcb45ae85f7376e0b7d3/runtime/compiler/control/JITClientCompilationThread.cpp#L892.

Reply:
ok, then alloca is fine. I was trying to avoid one round of copying by writing directly into the string buffer. It may work though if using  &mystring[0]

Reply:
`& nextSignatureString[0]` should work. Maybe it could be done in a separate PR since `alloca` is used at a few places where `getStringUTF8()` is called  in `JITClientCompilationThread.cpp` on constructing the message back to the server.